### PR TITLE
[Merged by Bors] - feat(Data.DFinsupp.Notation): add notation for `DFinsupp`

### PR DESCRIPTION
**Pull request by eric-wieser**
This just extends the existing notation we already use for `Finsupp` (#6367).
The two seem to coexist on the same notation without any issue.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/DFinsupp/Notation.lean [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace DFinsupp
+
+open Lean
+open Lean.Parser
+open Lean.Parser.Term
+
+extend_docs Finsupp.fun₀ after
+  "If the expected type is `Π₀ i, α i` (`DFinsupp`),
+  then this is notation for `DFinsupp.single` and  `Dfinsupp.updated` instead."
```
Comment (timotree3) [inline Mathlib/Data/DFinsupp/Notation.lean:29]:
```suggestion
  then this is notation for `DFinsupp.single` and  `Dfinsupp.update` instead."
```

File: Mathlib/Data/DFinsupp/Notation.lean [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace DFinsupp
+
+open Lean
+open Lean.Parser
+open Lean.Parser.Term
+
+extend_docs Finsupp.fun₀ after
+  "If the expected type is `Π₀ i, α i` (`DFinsupp`),
+  then this is notation for `DFinsupp.single` and  `Dfinsupp.update` instead."
```
Comment (eric-wieser) [inline Mathlib/Data/DFinsupp/Notation.lean:29]: This causes a panic in CI: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/PANIC.20at.20Lean.2EMapDeclarationExtension.2Einsert/near/423274327

**Review by joneugster — APPROVED** [review]
This looks like a very clean and short PR to me.

Hopefully it should be easy for a reviewer to delegate/merge!

Only question I have is whether there is any test-case for `Finsupp` which ensures the notation for `Finsupp` takes priority over `DFinsupp`, and if not, then maybe one could be added.

**Comment by eric-wieser** [comment]
> Only question I have is whether there is any test-case for `Finsupp` which ensures the notation for `Finsupp` takes priority over `DFinsupp`, and if not, then maybe one could be added.

Indeed, in the presence of both imports breaks things

**Comment by eric-wieser** [comment]
> Indeed, in the presence of both imports breaks things

This is now fixed.

**Review by YaelDillies — COMMENTED** [review]
Thanks, that's very nice!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**